src/Artifacts.jl

```diff
@@ -437,9 +446,22 @@ function ensure_artifact_installed(name::String, meta::Dict, artifacts_toml::Str
             download_success = with_show_download_info(io, name, quiet_download) do
                 download_artifact(hash, url, tarball_hash; verbose=verbose, quiet_download=quiet_download, io=io)
             end
-            download_success && return artifact_path(hash)
+            # download_success is either `true` or an error object
+            if download_success === true
+                return artifact_path(hash)
+            else
+                push!(errors, (url, download_success))
+            end
+        end
+        errmsg = """
+        Unable to automatically download/install artifact '$(name)' from sources listed in '$(artifacts_toml)'.
+        Sources attempted:\n
+        """
+        for (url, err) in errors
+            errmsg *= "$(url)\n"
+            errmsg *= "  Error: $(sprint(showerror, err))\n\n"
```

Comment:
Drop some whitespace?
```suggestion
        errmsg = """
        Unable to automatically download/install artifact '$(name)' from sources listed in '$(artifacts_toml)'.
        Sources attempted:
        """
        for (url, err) in errors
            errmsg *= "- $(url)\n"
            errmsg *= "    Error: $(sprint(showerror, err))\n"
```